### PR TITLE
Release/v0.5.0

### PR DESCRIPTION
**Pull request by kidunot89**
### Release Checklist
---------------------------------------------------
- [x] Support for **WPGraphQL** v0.8.0 added.
- [x] Deprecated queries removed.

### CODE REVIEW

File: includes/data/connection/class-downloadable-item-connection-resolver.php [hunk 1]
```diff
@@ -26,6 +27,32 @@ class Downloadable_Item_Connection_Resolver extends AbstractConnectionResolver {
...
+	public function get_loader_name() {
+		return 'downloadable_item';
+	}
+
+	/**
+	 * Given an ID, return the model for the entity or null
+	 *
+	 * @param $id
+	 *
+	 * @return
```
Comment (renatonascalves) [inline includes/data/connection/class-downloadable-item-connection-resolver.php]: Missing the return value here.

File: includes/data/connection/class-downloadable-item-connection-resolver.php [hunk 2]
```diff
@@ -26,6 +27,32 @@ class Downloadable_Item_Connection_Resolver extends AbstractConnectionResolver {
...
+	 * @return string
+	 */
+	public function get_loader_name() {
+		return 'downloadable_item';
+	}
+
+	/**
+	 * Given an ID, return the model for the entity or null
+	 *
+	 * @param $id
```
Comment (renatonascalves) [inline includes/data/connection/class-downloadable-item-connection-resolver.php]: Missing documentation of what this ID is.

File: includes/data/connection/class-downloadable-item-connection-resolver.php [hunk 3]
```diff
@@ -26,6 +27,32 @@ class Downloadable_Item_Connection_Resolver extends AbstractConnectionResolver {
 	 */
 	use WC_Connection_Functions;
 
+	const PREFIX = 'DI';
+
+	/**
+	 * get_loader_name
```
Comment (renatonascalves) [inline includes/data/connection/class-downloadable-item-connection-resolver.php]: Not sure why the name of the method is here.

File: includes/data/connection/class-downloadable-item-connection-resolver.php [hunk 4]
```diff
@@ -186,4 +215,18 @@ public function get_items() {
...
+	 * Validates Model.
+	 *
+	 * If model isn't a class with a `fields` member, this function with have be overridden in
+	 * the Connection class.
+	 *
+	 * @param array $model Downloadable item model.
+	 *
+	 * @return bool
+	 */
+	protected function is_valid_model( $model ) {
```
Comment (renatonascalves) [inline includes/data/connection/class-downloadable-item-connection-resolver.php]:
```suggestion
	protected function is_valid_model( array $model ) {
```

File: includes/data/connection/class-order-connection-resolver.php [hunk 5]
```diff
@@ -50,6 +51,26 @@ public function __construct( $source, $args, $context, $info ) {
...
+	 * @return string
+	 */
+	public function get_loader_name() {
+		return 'wc_cpt';
+	}
+
+	/**
+	 * Given an ID, return the model for the entity or null
+	 *
+	 * @param $id
```
Comment (renatonascalves) [inline includes/data/connection/class-order-connection-resolver.php]: Missing documentation...

File: includes/data/connection/class-product-connection-resolver.php [hunk 6]
```diff
@@ -66,6 +67,39 @@ public function __construct( $source, $args, $context, $info ) {
...
+	public function get_loader_name() {
+		return 'wc_cpt';
+	}
+
+	/**
+	 * Given an ID, return the model for the entity or null
+	 *
+	 * @param $id
+	 *
+	 * @return
```
Comment (renatonascalves) [inline includes/data/connection/class-product-connection-resolver.php]: Same here.

File: includes/data/loader/class-wc-cpt-loader.php [hunk 7]
```diff
@@ -134,35 +137,62 @@ function ( $split, \WP_Query $query ) {
...
-							function () use ( $post_type, $key ) {
-								return $this->resolve_model( $post_type, $key );
-							}
-						);
+			$context     = $this->context;
+			$customer_id = null;
+			$parent_id   = null;
+			
+			// Resolve post author for future capability checks.
+			if ( 'shop_order' === $post_type ) {
```
Comment (renatonascalves) [inline includes/data/loader/class-wc-cpt-loader.php]: Suggestion: I'd switch this conditional block of code for a `switch` statement.

File: includes/data/loader/class-wc-db-loader.php [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+						sprintf( __( 'Loader type invalid: %s', 'wp-graphql-woocommerce' ), $this->loader_type )
+					);
+				}
+		}
+
+		$loaded_items = array();
+
+		/**
+		 * Loop over the keys and return an array of cart items,
+		 */
```
Comment (renatonascalves) [inline includes/data/loader/class-wc-db-loader.php]:
```suggestion
		 // Loop over the keys and return an array of cart items.
```

File: includes/data/loader/class-wc-db-loader.php [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+			foreach ( $locales as $locale ) {
+				if ( empty( $rate->{'tax_rate_' . $locale->location_type} ) ) {
+					$rate->{'tax_rate_' . $locale->location_type} = array();
+				}
+				$rate->{'tax_rate_' . $locale->location_type}[] = $locale->location_code;
+			}
+			return new Tax_Rate( $rate );
+		} else {
+			return null;
+			// throw new UserError(
```
Comment (renatonascalves) [inline includes/data/loader/class-wc-db-loader.php]: Should this be here?

File: includes/data/loader/class-wc-db-loader.php [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+
+	/**
+	 * Returns the downloadable item connected the provided IDs.
+	 * 
+	 * @param int $id - Downloadable item ID.
+	 * 
+	 * @return WC_Customer_Download|null
+	 */
+	public function load_downloadable_item_from_id( $id ) {
+		$node = new \WC_Customer_Download( $id );
```
Comment (renatonascalves) [inline includes/data/loader/class-wc-db-loader.php]: I'd suggest importing this class to the top of the file for better visibility of the classes being used inside this file/class.

File: includes/data/loader/class-wc-db-loader.php [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+	 * 
+	 * @return Tax_Rate|null
+	 */
+	public function load_tax_rate_from_id( $id ) {
+		global $wpdb;
+
+		$rate = \WC_Tax::_get_tax_rate( $id, OBJECT );
+		if ( ! \is_wp_error( $rate ) && ! empty( $rate ) ) {
+			// Get locales from a tax rate.
+			$locales = $wpdb->get_results(
```
Comment (renatonascalves) [inline includes/data/loader/class-wc-db-loader.php]: Is there any value in caching this for a few minutes? If at all?

File: includes/connection/class-product-reviews.php [hunk 12]
```diff
@@ -46,7 +46,8 @@ public static function register_connections() {
 							'type'        => 'Float',
 							'description' => __( 'Review rating', 'wp-graphql-woocommerce' ),
 							'resolve'     => function( $source ) {
-								$rating = get_comment_meta( $source['node'], 'rating', true );
+								$review = $source['node'];
```
[the remark this answers is not in the document]
Comment (kidunot89) [reply includes/connection/class-product-reviews.php]: No, the edge can't exist without the `node`.

File: includes/data/connection/class-downloadable-item-connection-resolver.php (hunk 3 above)
Comment (kidunot89) [reply includes/data/connection/class-downloadable-item-connection-resolver.php]: It's carry over from when I copied the signature from WPGraphQL's AbstractConnectionResolver class. I'll clean it up.

File: includes/data/connection/class-downloadable-item-connection-resolver.php (hunk 4 above)
Comment (kidunot89) [reply includes/data/connection/class-downloadable-item-connection-resolver.php]: That's an overridden [function](https://github.com/wp-graphql/wp-graphql/blob/develop/src/Data/Connection/AbstractConnectionResolver.php#L557), adding typehinting actually breaks the class.

File: includes/data/loader/class-wc-db-loader.php (hunk 11 above)
Comment (kidunot89) [reply includes/data/loader/class-wc-db-loader.php]: What do you mean?

File: includes/data/loader/class-wc-db-loader.php (hunk 10 above)
Comment (kidunot89) [reply includes/data/loader/class-wc-db-loader.php]: Maybe. I'm completely dead set on this method of resolution of the `DownloadableItem` type. Seeing as it actually doesn't work for `DownloadableItem` connections, I actually bypass this method by **priming** the objects returned from the **$source** downloadable items getters define by WC on respective crud object [here](https://github.com/wp-graphql/wp-graphql-woocommerce/pull/249/files#diff-f6fe31218f292074ac3690e67c033ecbR146-R148).

**Pull request merged**